### PR TITLE
[FEATURE] Add TCA field control „passwordGenerator“

### PR DESCRIPTION
**Pull request by linawolf**
resolves https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/241, https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/243

### CODE REVIEW

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+Field control options
+---------------------
+
+title
+~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: title
+
+    :Path: $GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['title']
+    :Type: String / Localized String
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:101]:
```suggestion
    :Type: String / localized string
```

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+allowEdit
+~~~~~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: allowEdit
+
+    :Path: $GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['allowEdit']
+    :Type: boolean
+    :Default: :php:`true`
+
+    If set to :php:`false` the user cannot edit the generated password.
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:115]:
```suggestion
    If set to :php:`false`, the user cannot edit the generated password.
```

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+passwordRules.digitCharacters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: passwordRules.digitCharacters
+
+    :Path: $GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['passwordRules']['digitCharacters']
+    :Type: boolean
+    :Default: :php:`true`
+
+    If set to :php:`false` the generated password contains no digit.
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:169]:
```suggestion
    If set to :php:`false`, the generated password contains no digit.
```

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+passwordRules.lowerCaseCharacters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: passwordRules.lowerCaseCharacters
+
+    :Path: $GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['passwordRules']['lowerCaseCharacters']
+    :Type: boolean
+    :Default: :php:`true`
+
+    If set to :php:`false` the generated password contains no lower case characters.
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:181]:
```suggestion
    If set to :php:`false`, the generated password contains no lower case characters.
```

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+passwordRules.upperCaseCharacters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: passwordRules.upperCaseCharacters
+
+    :Path: :php:`$GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['passwordRules']['upperCaseCharacters']`
+    :Type: boolean
+    :Default: :php:`true`
+
+    If set to :php:`false` the generated password contains no upper case characters.
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:192]:
```suggestion
    If set to :php:`false`, the generated password contains no upper case characters.
```

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+passwordRules.specialCharacters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  t3-fieldcontrol-passwordgenerator:: passwordRules.specialCharacters
+
+    :Path: :php:`$GLOBALS['TCA'][$table]['columns'][$field]['config']['fieldControl']['passwordGenerator']['options']['passwordRules']['specialCharacters']`
+    :Type: boolean
+    :Default: :php:`false`
+
+    If set to :php:`true` the generated password also contains special
```
Comment (brotkrueml) [inline Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:203]:
```suggestion
    If set to :php:`true`, the generated password also contains special
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+..  include:: _Password_4.rst.txt
+
+
+Base64 random bytes, readonly
+------------------------------
+
+Example: `zrt8sJd6GiqUI_EFgjPiedOj--D0NbTVOJz`
+
+..  figure:: /Images/ManualScreenshots/PasswordGeneratorAllChars.png
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:82]: Yes I picked the wrong picture, exchanging it now. I think it is relevant as it shows the function of password generator where the password cannot be edited.

File: Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+Examples
+========
+
+Include special characters
+--------------------------
+
+Example: `tA7'jZEIv!{96z.D`
+
+..  figure:: /Images/ManualScreenshots/PasswordGeneratorAllChars.png
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:52]: I think it does not create confusion. developer know, how password generators work. Also pictures help navigate examples faster.

[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ColumnsConfig/Type/Password/Properties/PasswordGenerator.rst:52]: Then we should add the screenshot to every example and not just to two of them.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**